server/middleware/verifyJWT.js

```diff
@@ -0,0 +1,38 @@
+import jwt from "jsonwebtoken";
+
+export const verifyJWT = (req, res, next) => {
+  const authHeader = req.headers.token;
+
+  if (!authHeader) {
+    return res.status(401).send({ error: "Unauthorized" });
+  }
+
+  const token = authHeader.split(" ")[1];
+  jwt.verify(token, process.env.SECRET_ACCESS_KEY, (err, user) => {
+    if (err) {
+      return res.status(403).send({ error: "Invalid token" });
+    }
+    req.decoded = user;
+    next();
+  });
+};
+
+export const verifyTokenAndAuthorization = (req, res, next) => {
+  verifyJWT(req, res, () => {
+    if (req.decoded.id === req.params.id || req.decoded.isAdmin) {
+      next();
+    } else {
+      return res.status(403).send({ error: "You are not authorized" });
+    }
+  });
+};
```

Comment:
Logic is wrong please review. Check req.decoded is not undefined instead or you can remove and use the verifyJWT instead.